clients/src/main/java/org/apache/kafka/common/security/kerberos/LoginManager.java

```diff
@@ -36,14 +38,25 @@
     private final Login login;
     private final String serviceName;
     private final LoginType loginType;
+    private final Subject existingSubject;
     private int refCount;
 
     private LoginManager(LoginType loginType, Map<String, ?> configs) throws IOException, LoginException {
         this.loginType = loginType;
         String loginContext = loginType.contextName();
-        login = new Login(loginContext, configs);
+
+        // Check for an existing Subject
+        AccessControlContext context = AccessController.getContext();
+        existingSubject = context != null ? Subject.getSubject(context) : null;
+
+        // Otherwise try to login
+        if (existingSubject == null) {
```

Comment:
I think the requirement should be stronger, we should check the subject has a KerberosPrincipal to skip the kerberos login. Else we can run into a situation that there is an alternate login context that has nothing to do with Kerberos and we'll fail.